polar-core/src/vm.rs

```diff
@@ -1007,10 +982,10 @@ impl PolarVirtualMachine {
                     left: x,
                     right: right.clone(),
                 })?,
-                VariableState::Cycle(c) => self.isa_expr(&cycle_constraints(c), left, right)?,
-                VariableState::Partial(e) => self.isa_expr(&e, left, right)?,
+                _ => self.isa_expr(left, right)?,
             },
             (_, Value::Variable(r)) | (_, Value::RestVariable(r)) => match self.variable_state(r) {
+                // TODO (dhatch): Is this case tested?
```

Comment:
Hmmm, yes, I think so. Look at `test_non_instance_specializers`

I think the "legitimate" use case for this is something like:

```
allow(user: User { id: id, is_staff: True }, ... ) if ...
```

I.e you want the isa to bind variables in the pattern.